pkg/k8sclient/k8sclient.go

```diff
@@ -396,13 +396,14 @@ func TryLoadPodDelegates(pod *v1.Pod, conf *types.NetConf, clientInfo *ClientInf
 // InClusterK8sClient returns the `k8s.ClientInfo` struct to use to connect to
 // the k8s API.
 func InClusterK8sClient() (*ClientInfo, error) {
-	config, err := rest.InClusterConfig()
+	clientInfo, err := GetK8sClient("", nil)
 	if err != nil {
 		return nil, err
 	}
-
-	logging.Debugf("InClusterK8sClient: in cluster config: %+v", config)
-	return NewClientInfo(config)
+	if clientInfo == nil {
+		return nil, fmt.Errorf("failed to create in-cluster kube client")
+	}
+	return clientInfo, err
```

Comment:
@s1061123 just want to make sure this approach works with your incoming changes?